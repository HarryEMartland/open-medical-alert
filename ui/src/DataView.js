import React from 'react';
import ContactsView from "./ContactsView";
import ContactView from "./ContactView";
import AlertView from "./AlertView";
import MedicalAlertsView from "./MedicalAlertsView";

function DataView({data}) {
    return <div className='container'>
        <div className='row'>
            <div className='col pt-3'>
                <form>
                    <MedicalAlertsView data={data}/>
                    <ContactView {...data}/>
                    { data.emergencyContacts && <ContactsView emergencyContacts={data.emergencyContacts}/>}
                </form>
            </div>
        </div>

    </div>
}

export default DataView