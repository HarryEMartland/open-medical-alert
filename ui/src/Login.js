import React from 'react';
import ReCAPTCHA from "react-google-recaptcha";


function Login({onLogin, loading, setPassword, password, errorMessage, captchaChange, valid}) {

    return <div className='container-fluid'>
        <div className='row justify-content-md-center pt-5 text-center'>
            <div className='col-sm col-md-6 col-lg-4 col-xl-3'>
                <form onSubmit={onLogin} >
                    {errorMessage && <div className="alert alert-danger" role="alert">{errorMessage}</div>}
                    <div className="form-group">
                        <input type='password' className='form-control form-control-lg' disabled={loading} onChange={setPassword}
                               value={password} placeholder='Password'/>
                    </div>
                    <div className="form-group">
                        <ReCAPTCHA
                            sitekey={process.env.REACT_APP_CAPTCHA_SITE_KEY}
                            onChange={captchaChange}
                            onExpired={()=>{captchaChange()}}
                        />
                    </div>
                    <div className="form-group">
                        <input type='submit' className='btn btn-success btn-lg' disabled={!valid} value='Login'/>
                    </div>
                </form>
            </div>
        </div>
    </div>
}

export default Login