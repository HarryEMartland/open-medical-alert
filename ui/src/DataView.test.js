import {render} from "@testing-library/react";
import * as React from "react";
import Login from "./Login";

import exampleData from '../../server/data.example.json'
import DataView from "./DataView";

describe('Login', () => {

    it('should render with data', () => {

        const {container} = render(<DataView data={exampleData}/>);
        expect(container).toMatchSnapshot();

    })

    it('should render with empty data', () => {

        const {container} = render(<DataView data={{}}/>);
        expect(container).toMatchSnapshot();

    })
})