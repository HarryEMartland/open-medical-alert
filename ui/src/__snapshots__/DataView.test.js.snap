// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render with data 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col pt-3"
      >
        <form>
          <div>
            <div
              class="alert alert-danger"
            >
              I have diabetes (Type 2)
              <a
                class="close"
                href="https://www.nhs.uk/conditions/type-2-diabetes/"
                target="_blank"
              >
                <span
                  aria-hidden="true"
                >
                  ⓘ
                </span>
              </a>
            </div>
            <div
              class="alert alert-danger"
            >
              I have seizures
              <a
                class="close"
                href="https://www.nhs.uk/conditions/what-to-do-if-someone-has-a-seizure-fit/"
                target="_blank"
              >
                <span
                  aria-hidden="true"
                >
                  ⓘ
                </span>
              </a>
            </div>
            <div
              class="alert alert-danger"
            >
              I have hearing loss
              <a
                class="close"
                href="https://www.nhs.uk/conditions/hearing-loss/"
                target="_blank"
              >
                <span
                  aria-hidden="true"
                >
                  ⓘ
                </span>
              </a>
            </div>
            <div
              class="alert alert-danger"
            >
              I have Alzheimer's
              <a
                class="close"
                href="https://www.nhs.uk/conditions/alzheimers-disease/"
                target="_blank"
              >
                <span
                  aria-hidden="true"
                >
                  ⓘ
                </span>
              </a>
            </div>
          </div>
          <div>
            <div
              class="form-group"
            >
              <label>
                Name
              </label>
              <input
                class="form-control"
                disabled=""
                type="text"
                value="John Doe"
              />
            </div>
            <div
              class="form-group"
            >
              <label>
                Address
              </label>
              <textarea
                class="form-control"
                disabled=""
              >
                64 Zoo Lane, London
              </textarea>
            </div>
            <div
              class="form-group"
            >
              <label>
                Postcode
              </label>
              <input
                class="form-control"
                disabled=""
                type="text"
                value="D45 6ER"
              />
            </div>
          </div>
          <div>
            <h3>
              Emergency Contacts
            </h3>
            <div
              class="row pt-2"
            >
              <div
                class="col-6 pb-3"
              >
                <div
                  class="col card pt-3 pb-2"
                >
                  <div>
                    <div
                      class="form-group"
                    >
                      <label>
                        Name
                      </label>
                      <input
                        class="form-control"
                        disabled=""
                        type="text"
                        value="Jane Doe"
                      />
                    </div>
                    <div
                      class="form-group"
                    >
                      <label>
                        Address
                      </label>
                      <textarea
                        class="form-control"
                        disabled=""
                      >
                        64 Zoo Lane, London
                      </textarea>
                    </div>
                    <div
                      class="form-group"
                    >
                      <label>
                        Postcode
                      </label>
                      <input
                        class="form-control"
                        disabled=""
                        type="text"
                        value="D45 6ER"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="col-6 pb-3"
              >
                <div
                  class="col card pt-3 pb-2"
                >
                  <div>
                    <div
                      class="form-group"
                    >
                      <label>
                        Name
                      </label>
                      <input
                        class="form-control"
                        disabled=""
                        type="text"
                        value="Jane Doe"
                      />
                    </div>
                    <div
                      class="form-group"
                    >
                      <label>
                        Address
                      </label>
                      <textarea
                        class="form-control"
                        disabled=""
                      >
                        64 Zoo Lane, London
                      </textarea>
                    </div>
                    <div
                      class="form-group"
                    >
                      <label>
                        Postcode
                      </label>
                      <input
                        class="form-control"
                        disabled=""
                        type="text"
                        value="D45 6ER"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="col-6 pb-3"
              >
                <div
                  class="col card pt-3 pb-2"
                >
                  <div>
                    <div
                      class="form-group"
                    >
                      <label>
                        Name
                      </label>
                      <input
                        class="form-control"
                        disabled=""
                        type="text"
                        value="Jane Doe"
                      />
                    </div>
                    <div
                      class="form-group"
                    >
                      <label>
                        Address
                      </label>
                      <textarea
                        class="form-control"
                        disabled=""
                      >
                        64 Zoo Lane, London
                      </textarea>
                    </div>
                    <div
                      class="form-group"
                    >
                      <label>
                        Postcode
                      </label>
                      <input
                        class="form-control"
                        disabled=""
                        type="text"
                        value="D45 6ER"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Login should render with empty data 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col pt-3"
      >
        <form>
          <div />
          <div />
        </form>
      </div>
    </div>
  </div>
</div>
`;
